machines/run.html.md

```diff
@@ -0,0 +1,368 @@
+
+All Fly Machines are made from Docker images. When you `fly launch` an app, a Fly Launch scanner may generate one for you based on your source code, or it may be a prebuilt image you point to, or it may be built from your hand-crafted Dockerfile and source code.
+
+We're working more directly with Machines here, so your choices with `fly machine run` are "point to an image" or "point to a Dockerfile."
+
+### Build from a Dockerfile
+To build the image from a Dockerfile named `Dockerfile`, indicate the current working directory using the `<image>` argument.
+
+```cmd
+fly machine run .
+```
+
+Use the `--dockerfile` option to specify a Dockerfile with a different name. 
+
+```cmd
+fly machine run . --dockerfile Dockerfile-dev
+```
+
+### Use an existing image
+
+```cmd
+fly machine run ghcr.io/livebook-dev/livebook:0.11.4     
+```
+
+## Get a shell on a temporary Machine
+
+The following command creates a temporary Machine using the Dockerfile in the working directory, and logs you into an interactive shell on it: 
+
+```cmd
+fly machine run . --shell
+```
+
+If you don't specify an app, a temporary app is created for the Machine. When you log out of the shell, the Machine, and if applicable, the temporary app, is deleted.
+
+The default shell is Bash. The `--command` flag allows you to specify a different shell if Bash isn't present in the Machine's Docker image. Log in as a non-`root` user using the `--user` flag.
+
+If you just want a shell on a temporary Ubuntu Machine that's in your org's private network, omit the `<image>` argument:
+
+```cmd
+fly machine --shell
+```
+
+## Run with a custom ENTRYPOINT or CMD
+
+You can have the Fly Platform init override the ENTRYPOINT and CMD (if any) of the Machine's Docker image at startup.
+
+### Custom CMD
+
+Override CMD by including the command to run at the end of the `fly machine run` invocation. This example simply spins up a Debian Linux Machine with a `sleep` task to keep it awake; you can shell into it or whatever:
+
+```cmd
+fly machine run debian sleep inf
+```
+
+### Custom ENTRYPOINT
+
+Override ENTRYPOINT using `--entrypoint`. In this example we use the [`--file-local` option](/docs/machines/run#copy-a-local-file-into-the-machine-file-system) to send an entrypoint script to the Machine and the `--entrypoint` option to run the script:
+
+```cmd
+fly machine run debian --file-local /entrypoint.sh=./entrypoint.sh \
+                       --entrypoint "/entrypoint.sh" \
+                       sleep inf
+```
+
+Here's a trivial `entrypoint.sh` you can use with the above example:
+
+```bash
+#! /bin/bash
+
+echo "Hello from my Fly Machine"
+
+exec "$@"
+```
+
+## Choose the region
+
+Tell the Fly Platform which [region](/docs/reference/regions/) to create the Machine in with the `--region` flag; if for some reason it can't start a new Machine in that region, you'll get an error. If the `--region` flag is omitted, the platform chooses the nearest region to you.
+
+
+## Set hardware specs
```

Comment:
do we say hardware specs or machine resources here?

Reply:
I bet people will find it better with hardware specs, but machine resources is more accurate